pkg/ec2pricing/ec2pricing.go

```diff
@@ -154,25 +184,27 @@ func (p *EC2Pricing) GetOndemandInstanceTypeCost(instanceType string) (float64,
 		},
 	}
 
-	// Check cache first and return it if available
-	if price, ok := p.cache[instanceType]; ok {
-		return price, nil
-	}
-
 	pricePerUnitInUSD := float64(-1)
-	err := p.PricingClient.GetProductsPages(&productInput, func(pricingOutput *pricing.GetProductsOutput, nextPage bool) bool {
-		var err error
+	var processingErr error
+	// FIXME Only works for us-east-1 and ap-south-1
```

Comment:
this is fixed above in the constructor. The Pricing client is always initialized to us-east-1.